src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -1050,13 +1052,41 @@ private void AddValueInternal(DateTime time, Span<double> values, Span<byte> tag
             {
                 var timestampDiff = ((time - BaselineDate).Ticks / 10_000);
                 var inRange = timestampDiff < int.MaxValue;
-                if (inRange == false || segment.Append(_context.Allocator, (int)timestampDiff, values, tagSlice, status) == false)
+                if (inRange == false ||  segment.Append(_context.Allocator, (int)timestampDiff, values, tagSlice, status) == false)
                 {
-                    FlushCurrentSegment(ref segment, values, tagSlice, status);
+                    var segmentLastTimestamp = segment.GetLastTimestamp(BaselineDate);
+                    if (segmentLastTimestamp == BaselineDate && time == segmentLastTimestamp)// all the entries in the segment has the same timestamp
+                    {
+                        if (FromReplication == false)
+                            throw new InvalidDataException(
+                                $"Segment reached to capacity and cannot receive more values with {time} timestamp on time series {_name} for {_docId} . " +
+                                "You may choose a different timestamp.");
+
+                        MergeAllEntriesAtSameTimestampToSingleEntry(ref segment, tagSlice, values);
```

Comment:
Need to rename this method. It no longer merges, it discards and leave just a single value.

Reply:
Done